dark/sam.py

```diff
@@ -1262,3 +1260,233 @@ def getReferenceInfo(
             )
 
     return bamId, reference, referenceLength
+
+
+class ReferenceReads:
+    """
+    Hold information about the reads that match a reference.
+
+    @param id_: A C{str} reference id.
+    @param length: The C{int} length of the reference.
+    """
+
+    def __init__(self, id_: str, length: int) -> None:
+        assert length
+        self.id_ = id_
+        self.length = length
+        self.coveredOffsets: set = set()
+        self.duplicate: set = set()
+        self.nonDuplicate: set = set()
+        self.primary: set = set()
+        self.qcFail: set = set()
+        self.readIds: set = set()
```

Comment:
Yes, in principle I agree. But... I don't like properties that much, because they hide things. Also, doing it that way relies on the logic of the `add` function more than just unconditionally adding all ids to a set. Probably not a big deal since something is either a duplicate or it's not, but I like it less.

The only time I use properties is when I have something that starts life as a regular object attribute and then I make it into a function. To keep backward compatibility I then make it a property.